### PR TITLE
docs: added dependency info in csharp

### LINKED ISSUE #446: C# presets uses .NET dependencies
There is no clear documentation about which presets uses .NET dependencies, and I did not even check it myself when creating them.

We need some clear structure so users know what presets they can use.

### PR DESCRIPTION
**Pull request by Samridhi-98**
**Description**

- Updated `docs/language/Csharp.md`

**Related issue(s)**

Fixes #446

### CODE REVIEW

**Comment by Samridhi-98** [comment]
@jonaslagoni  please review 🙂

File: docs/languages/Csharp.md [hunk 1]
```diff
@@ -15,6 +15,15 @@ There are special use-cases that each language supports; this document pertains
...
 Sometimes you want to serialize the data models into JSON. In order to do that use the preset `CSHARP_JSON_SERIALIZER_PRESET`
 
+**Important Note:**
+The following dependencies used in the serializer and deserializer functionality are .NET specific dependencies.
+
+```
+System.Text.Json
+System.Text.Json.Serialization
+System.Text.RegularExpressions
+```
```
Comment (jonaslagoni) [inline docs/languages/Csharp.md:25]: Would it make sense to also give direct links to the dependencies? Such as https://github.com/asyncapi/modelina/pull/596/files#diff-9dd59a8eea1ff041d2dd3864867d10d85bf26318ff2848e23245a7f4a113d67cR67-R68 ?

Comment (Samridhi-98) [reply docs/languages/Csharp.md:25]: Yeah, It's a nice idea I will update it.

File: docs/languages/Csharp.md [hunk 2]
```diff
@@ -15,6 +15,17 @@ There are special use-cases that each language supports; this document pertains
...
```
Comment (jonaslagoni) [inline docs/languages/Csharp.md:25]: Now that we have `External dependencies` do you think we need `Important Note` as well? 🤔 They are kinda saying the same thing 🙂

Comment (Samridhi-98) [reply docs/languages/Csharp.md]: Yeah, corrected.

**Review by jonaslagoni — APPROVED** [review]
LGTM, awesome @Samridhi-98 🎉

**Pull request merged**